### PR TITLE
Curation - Recognize 403 or 500 errors from the tree builder

### PR DESCRIPTION
**Pull request by asafambar**
- [X] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [X] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [X] This pull request is on the dev branch.
- [X] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Comment by asafambar** [comment]
This PR handle potential errors that can occur while building mvn tree without configuring the "pass-through' configuration on the curated repo, in this case we want to prompt to the user a msg that inform him to check the curated repos configured correctly.

File: utils/java/mvn_test.go [hunk 1]
```diff
@@ -344,3 +344,43 @@ func TestRemoveMavenConfig(t *testing.T) {
...
+				"The following artifacts could not be resolved: junit:junit:pom:3.8.1 (absent): Could not transfer artifact junit:junit:pom:3.8.1 " +
+				"from/to artifactory (http://test:8046/artifactory/api/curation/audit/maven-remote): status code: 403, reason phrase: Forbidden (403)",
+		},
+		{
+			name:          "failed on 500",
+			wantMsgToUser: "Please verify pass-through enabled on the curated repos",
+			input: "[ERROR] Failed to execute goal on project my-app: Could not resolve dependencies for project com.mycompany.app:my-app:jar:1.0-SNAPSHOT: " +
+				"Failed to collect dependencies at junit:junit:jar:3.8.1: Failed to read artifact descriptor for junit:junit:jar:3.8.1: The following artifacts" +
+				" could not be resolved: junit:junit:pom:3.8.1 (absent): Could not transfer artifact junit:junit:pom:3.8.1 from/to artifactory" +
+				" (http://test:8046/artifactory/api/curation/audit/virtual-to-smart): status code: 500, reason phrase: Internal Server Error (500)",
```
Comment (attiasas) [inline utils/java/mvn_test.go:368]: If you are already splitting the `input` and concatatane multiple strings, you have similar entries and you can merge them as const and concatanate thier values for readability.

File: utils/java/mvn_test.go [hunk 2]
```diff
@@ -344,3 +344,43 @@ func TestRemoveMavenConfig(t *testing.T) {
...
+				"collect dependencies at junit:junit:jar:3.8.1: Failed to read artifact descriptor for junit:junit:jar:3.8.1: " +
+				"The following artifacts could not be resolved: junit:junit:pom:3.8.1 (absent): Could not transfer artifact junit:junit:pom:3.8.1 " +
+				"from/to artifactory (http://test:8046/artifactory/api/curation/audit/maven-remote): status code: 400, reason phrase: Forbidden (400)",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mdt := &MavenDepTreeManager{}
+			msg := mdt.suspectCurationBlockedError(tt.input)
+			assert.Contains(t, tt.wantMsgToUser, msg)
```
Comment (attiasas) [inline utils/java/mvn_test.go:383]:
```suggestion
			assert.Contains(t, tt.wantMsgToUser, mdt.suspectCurationBlockedError(tt.input))
```

File: utils/java/mvn.go [hunk 3]
```diff
@@ -246,3 +251,12 @@ func (mdt *MavenDepTreeManager) CreateTempDirWithSettingsXmlIfNeeded() (tempDirP
...
 	return
 }
+
+// In case mvn tree fails on 403 or 500 it can be related to packages blocked by curation.
+// For this use case to succeed, pass through should be enabled in the curated repos
+func (mdt *MavenDepTreeManager) suspectCurationBlockedError(cmdOutput string) (msgToUser string) {
+	if mdt.isCurationCmd && (strings.Contains(cmdOutput, "status code: 403") || strings.Contains(cmdOutput, "status code: 500")) {
+		msgToUser = "Failed to get dependencies tree for maven project, Please verify pass-through enabled on the curated repos"
+	}
+	return msgToUser
```
Comment (attiasas) [inline utils/java/mvn.go:261]:
```suggestion
    if !mdt.isCurationCmd {
        return
    }
    if strings.Contains(cmdOutput, "status code: 403") || strings.Contains(cmdOutput, "status code: 500") {
        msgToUser = "Failed to get dependencies tree for maven project, Please verify pass-through enabled on the curated repos"
    }
    return
```

**Review by attiasas — CHANGES_REQUESTED** [review]
Please add to the description how the log will look in case of those error

File: utils/java/mvn_test.go (hunk 1 above)
Comment (asafambar) [reply utils/java/mvn_test.go:368]: its a test, just copied the mvn cli output

**Pull request merged**